### PR TITLE
search functionality finish

### PR DESCRIPTION
**Pull request by saidarshan27**
1. Search bar added. 
2. when the search bar gets empty, the state is reset to show all nft cards. 
3. I have moved the API calling logic entirely to `NftList.js`. This will avoid `NftCard` component calling API inside a loop, which in itself is a performance issue.
4. Changed props for `NftCard` to receive all the required parameters to render, rather just imgHash. 
5. Added a loading text to enhance user experience. 

Please review the code and get back to me in the review comments in case of any changes. I will be happy to make those changes.

### CODE REVIEW

**Comment by SahityaRoy** [comment]
@bhupendra-chouhan @neutr0n420  have look into this PR.

**Comment by bhupendra-chouhan** [comment]
> @bhupendra-chouhan @neutr0n420 have look into this PR.

Sure!

File: src/components/Body/NftList/NftList.js [hunk 1]
```diff
@@ -19,6 +20,40 @@ const contract = getContract({
...
+
+  const onSearchBtnClick = () => {
+    try {
+      if (searchQuery?.length) {
+        const trimmedSearchVal = searchQuery.trim();
+    
+        const filteredResults = nftList.filter((nft) => {
+          const githubId = nft.attributes[1].value;
+    
+          return githubId.includes(trimmedSearchVal);
```
Comment (bhupendra-chouhan) [inline src/components/Body/NftList/NftList.js:34]: Hey @saidarshan27  you did a good work. Let's refine your contribution a bit more.

## Contribution Request: Remove Case Sensitivity in Search Bar Input
- [ ] We would like to improve the search functionality by removing case sensitivity. The goal is to ensure that the search works regardless of whether users input uppercase or lowercase letters. Please modify the search logic to handle all inputs uniformly by converting both the search query and the searchable data to the same case (either lowercase or uppercase) before performing the search.

- [ ] Before submitting your pull request, please ensure that your forked repository is in sync with the latest code from the main branch of this repository. Here are the steps to do so:

**Review by bhupendra-chouhan — CHANGES_REQUESTED** [review]

File: src/components/Body/NftList/NftList.js (hunk 1 above)
Comment (saidarshan27) [reply src/components/Body/NftList/NftList.js:35]: Hey @bhupendra-chouhan ,

I have made the search functionality to be case insensitive. Also I have added a Loader component.

The loader component accepts a type prop: 'inline','partial','full'
As the names suggest, inline type loader can be used inside buttons.
partial is not a full screen loader without the background.
full is full screen loader, which will overlay the entire screen.

You can also control the loader color as well as the background color in `loader.css`

**Comment by bhupendra-chouhan** [comment]
👏 Awesome work @saidarshan27  . Your PR has been successfully merged.

**Pull request merged**